### PR TITLE
The first iteration of the API.

### PR DESCRIPTION
**Pull request by Djis247**
The backend is a WIP, but the API supports a GET request to /api/person/123 to retrieve details about person 123. It also supports a POST request to /api/person to create a person and returns that persons id.

### CODE REVIEW

**Comment by Djis247** [comment]
BTW... before using the API in React, you should setup React to proxy the requests to the backend. [Here is a link that explains it in more detail.](https://medium.com/bb-tutorials-and-thoughts/react-how-to-proxy-to-backend-server-5588a9e0347)

File: server/dbSchema.js [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+let handler = (err)=> {
+	if (err) console.error(err);
+}
+db.exec("PRAGMA foreign_keys = ON;")
+db.exec("CREATE TABLE IF NOT EXISTS people(\
+	id			INTEGER		PRIMARY KEY,\
+	firstName	TEXT	NOT NULL,\
+	lastName	TEXT	NOT NULL,\
+	birthDate	TEXT	NOT NULL,\
+	awanaGroup	INTEGER		DEFAULT NULL,\
```
Comment (davidviso) [inline server/dbSchema.js:12]: awanaGroup should be TEXT, INT saves space and whatnont but TEXT is more explicit and the frontend already uses strings

File: server/dbSchema.js [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+	person		INTEGER		UNIQUE,\
+	contact		INTEGER,\
+	FOREIGN KEY(person)		REFERENCES people(id),\
+	FOREIGN KEY(contact)	REFERENCES contacts(id)\
+)", handler)
+db.exec("CREATE TABLE IF NOT EXISTS awanaGroups(\
+	id			INTEGER		PRIMARY KEY,\
+	name		TEXT	NOT NULL\
+)", handler)
+db.get("SELECT COUNT(id) AS total FROM awanaGroups", (err, row)=>{
```
Comment (davidviso) [inline server/dbSchema.js:35]: or maybe I just don't know how db works

**Review by davidviso — APPROVED** [review]

File: server/dbSchema.js (hunk 1 above)
Comment (Djis247) [reply server/dbSchema.js:12]: there is an awanaGroup table that it references.. this design allows for saving space and preventing the use of strings that don't correspond to an approved group

**Pull request merged**